maas/client/viscera/machines.py

```diff
@@ -192,7 +199,9 @@ class MachinesType(NodesType):
             'cpu_count': str(cpus) if cpus else None,
             'mem': str(memory) if memory else None,
             'pod': pod,
```

Comment:
I think you actually need to check the type. If its a Pod use the ID from the object.